libs/langchain/langchain/document_loaders/airbyte_cdk.py

```diff
@@ -0,0 +1,27 @@
+"""Loads local airbyte json files."""
+import json
+from typing import Any, Iterator, List, Mapping, Optional
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+from airbyte_protocol.models.airbyte_protocol import AirbyteRecordMessage, AirbyteStateMessage
+from airbyte_cdk.sources.embedded.base_integration import BaseEmbeddedIntegration
+from airbyte_cdk.sources.embedded.runner import CDKRunner
+
+
+class AirbyteCDKLoader(BaseLoader, BaseEmbeddedIntegration):
+    """Loads records using an Airbyte source connector implemented using the CDK."""
+
+    def __init__(self, config: Mapping[str, Any], source_class: Any, stream_name: str, state: Optional[AirbyteStateMessage] = None) -> None:
+        super().__init__(config=config, runner=CDKRunner(source=source_class(), name=source_class.__name__))
+        self._stream_name = stream_name
+        self._state = state
+
+    def _handle_record(self, record: AirbyteRecordMessage, id: Optional[str]) -> Document:
+        return Document(page_content="", metadata=record.data)
+
+    def load(self) -> List[Document]:
+        return list(self._load_data(stream_name=self._stream_name, state=self._state))
+
+    def lazy_load(self) -> Iterator[Document]:
+        return self._load_data(stream_name=self._stream_name, state=self._state)
```

Comment:
I notice we're passing state through... should the loader also internally update `_state` with the latest data, or do we want to make it explicit for users to pass through state themselves?

Just thinking about the ergonomics, someone that wants to implement incremental would need to instantiate a new instance of the loader if state changed. Not sure if instead we should pass state in the `load` method rather than on init.

Thinking about this from the Airbyte PoV, state changes on each run of the source, but the other info like configs/catalog stays the same throughout, which is why I'm leaning a bit towards moving state to the load.

Reply:
Following the slack convo, if this is part of the langchain interface please ignore me 😛